src/manifolds/VectorBundle.jl

```diff
@@ -943,10 +1073,43 @@ function vector_transport_direction(M::VectorBundle, p, X, d)
     return vector_transport_direction(M, p, X, d, M.vector_transport)
 end
 
+function _vector_transport_direction(
+    M::VectorBundle,
+    p,
+    X,
+    d,
+    m::VectorBundleVectorTransport,
```

Comment:
For consistency this could also get the Product in its name?

Reply:
We could, though technically it would be breaking. `VectorBundleVectorTransport` has existed for a long time.

Reply:
We could also introduce an additional `VectorBundleProductVectorTransport` and not break things ;)

Reply:
You mean as an alias?

Reply:
or as a second type.

Reply:
But would it do anything differently? If not, should the old one be deprecated?

Reply:
I am not 100% sure what the old one did, since now it is a new implementation, is the old one is now useless, then one could deprecate it, yes. If it has some use we just could keep both?

Reply:
This is the old one, it had `vector_transport_to` defined before. I've just added `vector_transport_direction` here.

Reply:
Hm, that was chaotic then, because someone (me?) implemented parallel transport (to?) as the component wise wile someone (you?) did the same using a retraction.

Good that it is now unified. But then I would prefer to use `Product` within that for now, unless it makes sense to omit `Product` for all three, retract/inverse retract/vector transport, because it gets clear that the VectorBundleX is the one meant element wise.
If that gets not clear, all should get the Product in their name.

Reply:
OK, I'll add `Product` then, and maybe keep the old name as a deprecated alias?

Reply:
that would be my preferred way, yes.

Reply:
OK, done. There is a `Base.@deprecate_binding` which I was trying to use instead of a constant but it doesn't seem to do anything, i.e. I could still use the old binding without any warnings in REPL.

Reply:
at least in my changes here line 1081 still does not have the `Product` inside, is that a bug in GitHub or did you miss to update that?

Reply:
Line 1081 is the empty line between two methods of `vector_space_dimension` so probably a Github issue :slightly_smiling_face: .